locust/test/subprocess_utils.py

```diff
@@ -0,0 +1,228 @@
+import os
+import re
+import shlex
+import signal
+import subprocess
+import time
+from collections.abc import Callable
+from typing import IO, Any
+
+import gevent
+import psutil
+import pytest
+
+from .util import IS_WINDOWS
+
+
+class TestProcess:
+    """
+    Wraps a subprocess for testing purposes.
+    """
+
+    __test__ = False
+    process_backend = subprocess
+    proc: subprocess.Popen[str]
+
+    def __init__(
+        self,
+        command: str,
+        on_fail: Callable[[str], None] = pytest.fail,
```

Comment:
I dont think we're passing on_fail anywhere anymore? removing it will simplify this class a lot.